main.go

```diff
@@ -52,131 +53,117 @@ depend on another.
-	options := &ViewOptions{
-		ShowAll:   *allFlag,
-		Summarise: *summaryFlag,
-		Order:     order,
-		Reversed:  reversed,
-	}
+func doView(tasks TaskList, options *ViewOptions) {
 	view := NewConsoleView()
 	view.ShowTree(tasks, options)
 }
 
-func doAdd(tasks TaskList, graft TaskNode, priority Priority, text string) {
+func doAdd(tasks TaskList, graft TaskNode, priority Priority, text string, config *config) {
 	graft.Create(text, priority)
-	saveTaskList(tasks)
+	saveTaskList(tasks, config)
 }
 
-func doEditTask(tasks TaskList, task Task, priority Priority, text string) {
+func doEditTask(tasks TaskList, task Task, priority Priority, text string, config *config) {
 	if text != "" {
 		task.SetText(text)
 	}
 	if priority != -1 {
 		task.SetPriority(priority)
 	}
-	saveTaskList(tasks)
+	saveTaskList(tasks, config)
 }
 
-func doMarkDone(tasks TaskList, references []Task) {
+func doMarkDone(tasks TaskList, references []Task, config *config) {
 	for _, task := range references {
 		task.SetCompleted()
 	}
-	saveTaskList(tasks)
+	saveTaskList(tasks, config)
 }
 
-func doMarkNotDone(tasks TaskList, references []Task) {
+func doMarkNotDone(tasks TaskList, references []Task, config *config) {
 	for _, task := range references {
 		task.SetCompletionTime(time.Time{})
 	}
-	saveTaskList(tasks)
+	saveTaskList(tasks, config)
 }
 
-func doReparent(tasks TaskList, task TaskNode, below TaskNode) {
+func doReparent(tasks TaskList, task TaskNode, below TaskNode, config *config) {
 	ReparentTask(task, below)
-	saveTaskList(tasks)
+	saveTaskList(tasks, config)
 }
 
-func doRemove(tasks TaskList, references []Task) {
+func doRemove(tasks TaskList, references []Task, config *config) {
 	for _, task := range references {
 		task.Delete()
 	}
-	saveTaskList(tasks)
+	saveTaskList(tasks, config)
 }
 
-func doPurge(tasks TaskList, age time.Duration) {
+func doPurge(tasks TaskList, age time.Duration, config *config) {
 	cutoff := time.Now().Add(-age)
 	matches := tasks.FindAll(func(task Task) bool {
 		return !task.CompletionTime().IsZero() && task.CompletionTime().Before(cutoff)
 	})
 	for _, m := range matches {
 		m.Delete()
 	}
-	saveTaskList(tasks)
+	saveTaskList(tasks, config)
 }
 
-func doSetTitle(tasks TaskList, args []string) {
+func doSetTitle(tasks TaskList, args []string, config *config) {
 	title := strings.Join(args, " ")
 	tasks.SetTitle(title)
-	saveTaskList(tasks)
+	saveTaskList(tasks, config)
 }
 
-func doShowInfo(tasks TaskList, index string) {
+func doShowInfo(tasks TaskList, index string, options *ViewOptions) {
 	task := tasks.Find(index)
 	if task == nil {
 		fatalf("no such task %s", index)
 	}
 	view := NewConsoleView()
-	view.ShowTaskInfo(task)
+	view.ShowTaskInfo(task, options)
 }
 
-func processAction(tasks TaskList) {
-	priority := PriorityFromString(*priorityFlag)
+func processAction(tasks TaskList, config *config) {
+	priority := PriorityFromString(config.Priority)
 	var graft TaskNode = tasks // -golint
-	if *graftFlag != "root" {
```

Comment:
IMO this is mixing concerns slightly, as `--graft` is not really a configuration option, it's an action, so it shouldn't really be in the config file. OTOH passing around a config object makes life cleaner, so I think it's fine as-is.